mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/property/NoConnectionProvisioningModelProperty.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.property;
+
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.extension.api.ExtensionConstants;
+
+/**
+ * Marker {@link ModelProperty} for connectable components that require a connection yet connection provisioning needs to be
+ * skipped.
+ *
+ * @since 1.5
+ */
+public class NoConnectionProvisioningModelProperty implements ModelProperty {
```

Comment:
by looking at the comments in this PR and the corresponding mule one I think that this modelProperty should either not exist (ideal) or at the very least be redefined as what you're really modeling is not the provisioning (which is an implementation detail that could change at any time in any SDK) but the provenance. This MP also implies that an operation uses one single connection, which is true in Java but not in Mule SDK

Reply:
I removed this MP and added a new one which aims to represent the SDK flavor used for developing the extension.
Please let me know if the name for that one is good enough (`SdkFlavorModelProperty`).

Also I only added the model property on the major flavors like Java SDK, XML SDK, Mule SDK, Mule SDK (in app).

I know there may be extensions that are not generated by any of these (for example the ones declared programmatically). Maybe there are other flavors I didn't know of. And also I think there is the possibility to contribute `ExtensionLoadingDelegates` by using the `CraftedExtensionModelLoader`.

Do you think it is ok to just leave out the flavor model property for those cases? such situation could be treated as "unknown" for the time being.

Since this is a private model property for now, we have room for improving those kind of areas if/when we find the justification to do so.